src/wazuh_modules/wm_database.c

```diff
@@ -532,6 +554,8 @@ int wm_sync_agentinfo(int id_agent, const char *path) {
                     os_platform ++;
                 }
             }
+            os_arch = (char *) malloc(50);
+            wm_get_os_arch(os, os_arch);
```

Comment:
As I suggested above, you may replace both lines for:
```c
os_arch = wm_get_os_arch(os);
```

The memory will be allocated in the same way as with `malloc()`. No further changes –like `free()`– are needed.